theano/tensor/subtensor.py

```diff
@@ -1082,6 +1082,8 @@ def inc_subtensor(x, y, inplace=False, set_instead_of_inc=False,
     if not x.owner:
         raise TypeError('x must be the result of a subtensor operation')
 
+    if isinstance(x.owner.op, theano.printing.Print):
+        x = x.owner.inputs[0]
```

Comment:
You would have to keep a reference to that `Print` Op, and apply it on the new Op later.
For instance, rebuild a Print op that gets executed after the AdvancedSubtensor, and prints the value of the original Subtensor.